assets/javascripts/discourse/components/chat-message-collapser.js

```diff
@@ -40,4 +41,38 @@ export default Component.extend({
   close() {
     this.set("collapsed", true);
   },
+
+  @computed("cooked")
+  get isYoutube() {
+    return isYoutube(this.cooked);
+  },
+
+  @computed("uploads", "cooked")
+  get isImage() {
+    return isImage(this.uploads, this.cooked);
+  },
+
+  @computed("cooked", "message")
+  get isAnimatedImage() {
+    return isAnimatedImage(this.cooked, this.message);
+  },
 });
+
+function isYoutube(cooked) {
+  return /^<div class="onebox lazyYT lazyYT-container"/.test(cooked);
+}
+
+function isAnimatedImage(cooked, message) {
```

Comment:
This is slightly misnamed, you're targeting animated onebox images here. I think it might be good to change this to `isOneboxedImage` and target all `img` elements with a onebox class, and let them be collapsible.

Reply:
Also, nitpick, but you likely don't need to pass the message, instead, like with YouTube above you can search for an img tag and a onebox class.

Reply:
Actually for this commit, I contemplated using `message` for the youtube link instead of deriving it from the onebox :|

Reply:
> target all `img`

I think that's good for the next PR. I just tried with https://i.guim.co.uk/img/media/c5e73ed8e8325d7e79babf8f1ebbd9adc0d95409/2_5_1754_1053/master/1754.jpg?width=465&quality=45&auto=format&fit=max&dpr=2&s=065f279099ded1062688e357b155dc29 and it doesn't show up properly